sys/fuchsia/fidl.txt

```diff
@@ -5,10 +5,10 @@
 
 include <zircon/fidl.h>
 
-type fidl_call_args[REQ_MESSAGE, REQ_HANDLES, RESP_HANDLES] {
+type fidl_call_args[REQ_MESSAGE, REQ_HANDLES, RESP_MESSAGE, RESP_HANDLES] {
 	wr_bytes	ptr[in, REQ_MESSAGE]
 	wr_handles	ptr[in, REQ_HANDLES]
-	rd_bytes	ptr[out, array[int8]]
+	rd_bytes	ptr[out, RESP_MESSAGE]
```

Comment:
As far as I understand, clients cannot determine the size for variable-size responses in advance. Instead, it will provide a buffer of maximum size (`ZX_CHANNEL_MAX_MSG_BYTES`), or an expected size, if any. It does not need to be pre-filled.

I do not see immediate reason to detail response messages. So the latest change (i) makes response messages untyped, and (ii) fixes their size to `ZX_CHANNEL_MAX_MSG_BYTES`. (see the arguments of zx_channel_call/zx_channel_read) We can detail response messages later, if necessary.